src/lib/queue.ts

```diff
@@ -73,7 +81,7 @@ export class Queue<K extends KubernetesObject> {
    *
    * @returns A promise that resolves when the webapp is reconciled
    */
-  async #dequeue() {
+  async #dequeue(): Promise<false | undefined> {
```

Comment:
returning `false | undefined` makes me wonder about where this is consumed. There's the possibility for some unexpected behavior in conditional statements.
> `false`: Is falsy but has a defined type and value.
> `undefined`: Is also falsy but indicates the absence of a value.

Do you think this justifies a small follow-up issue to dig into how this is consumed? It's definitely out-of-scope on this PR focused on just _adding_ return types.

Reply:
as the queue is such a critical piece i definitely think it deserves a close look

Reply:
https://github.com/defenseunicorns/pepr/issues/1530